### ISSUE TITLE
Karaf deployment issue

### ISSUE THREAD

**Issue by KevinPhemius** [fp7-netide/Engine#19]
Hello,

when I try to follow the deployment steps, I have an error:

<pre>
karaf@root()> feature:repo-add mvn:eu.netide.core/core/1.0.0.0/xml/features
Adding feature url mvn:eu.netide.core/core/1.0.0.0/xml/features
karaf@root()> feature:install netide-core
karaf@root()> DefaultValidationEventHandler: [FATAL_ERROR]: Premature end of file. 
     Location: 
</pre>

The karaf log show that an .xml file for the composition module is not found:

<pre>
karaf@root()> log:display
2015-08-21 11:25:03,718 | INFO  | pool-2-thread-1  | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |   org.apache.karaf.bundle.blueprintstate/4.0.1
2015-08-21 11:25:03,719 | INFO  | pool-26-thread-1 | CommandExtension                 | 43 - org.apache.karaf.shell.core - 4.0.1 | Registering commands for bundle org.apache.karaf.log.core/4.0.1
2015-08-21 11:25:03,734 | INFO  | pool-2-thread-1  | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 | Done.
2015-08-21 11:25:35,530 | INFO  | nsole user karaf | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 | Adding features: netide-core/[0,0.0.0]
2015-08-21 11:25:35,754 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 | Changes to perform:
2015-08-21 11:25:35,757 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |   Region: root
2015-08-21 11:25:35,757 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |     Bundles to install:
2015-08-21 11:25:35,757 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |       mvn:eu.netide.core/core.api/1.0.0.0
2015-08-21 11:25:35,758 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |       mvn:eu.netide.core/core.caos/1.0.0.0
2015-08-21 11:25:35,758 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |       mvn:eu.netide.core/core.connectivity/1.0.0.0
2015-08-21 11:25:35,758 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |       mvn:eu.netide.core/core.logpub/1.0.0.0
2015-08-21 11:25:35,758 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |       mvn:eu.netide.core/core.management/1.0.0.0
2015-08-21 11:25:35,758 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |       mvn:eu.netide.lib/netip/1.0.0.0
2015-08-21 11:25:35,760 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 | Installing bundles:
2015-08-21 11:25:35,760 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |   mvn:eu.netide.core/core.api/1.0.0.0
2015-08-21 11:25:35,805 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |   mvn:eu.netide.core/core.caos/1.0.0.0
2015-08-21 11:25:35,808 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |   mvn:eu.netide.core/core.connectivity/1.0.0.0
2015-08-21 11:25:35,812 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |   mvn:eu.netide.core/core.logpub/1.0.0.0
2015-08-21 11:25:35,814 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |   mvn:eu.netide.core/core.management/1.0.0.0
2015-08-21 11:25:35,817 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |   mvn:eu.netide.lib/netip/1.0.0.0
2015-08-21 11:25:36,482 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 | Starting bundles:
2015-08-21 11:25:36,483 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |   netip/1.0.0.0
2015-08-21 11:25:36,484 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |   core.api/1.0.0.0
2015-08-21 11:25:36,495 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |   core.logpub/1.0.0.0
2015-08-21 11:25:36,496 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |   core.caos/1.0.0.0
2015-08-21 11:25:36,530 | INFO  | pool-27-thread-1 | BlueprintContainerImpl           | 12 - org.apache.aries.blueprint.core - 1.4.3 | Bundle core.caos is waiting for dependencies [(objectClass=eu.netide.core.api.IShimManager)]
2015-08-21 11:25:36,530 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |   core.management/1.0.0.0
2015-08-21 11:25:36,573 | INFO  | tor Recieve Loop | ZeroMQBasedManagementConnector   | 55 - core.management - 1.0.0.0 | ZeroMQBasedManagementConnector started.
2015-08-21 11:25:36,579 | INFO  | pool-27-thread-1 | ManagementHandler                | 55 - core.management - 1.0.0.0 | ManagementHandler started.
2015-08-21 11:25:36,586 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 |   core.connectivity/1.0.0.0
2015-08-21 11:25:36,602 | INFO  | tor Recieve Loop | ZeroMQBasedManagementConnector   | 55 - core.management - 1.0.0.0 | Listening on port 5556
2015-08-21 11:25:36,628 | INFO  | tor Receive Loop | ZeroMQBasedConnector             | 53 - core.connectivity - 1.0.0.0 | ZeroMQBasedConnector started.
2015-08-21 11:25:36,629 | INFO  | pool-27-thread-1 | ShimManager                      | 53 - core.connectivity - 1.0.0.0 | ShimManager started.
2015-08-21 11:25:36,647 | INFO  | pool-27-thread-1 | FeaturesServiceImpl              | 8 - org.apache.karaf.features.core - 4.0.1 | Done.
2015-08-21 11:25:36,682 | INFO  | tor Receive Loop | ZeroMQBasedConnector             | 53 - core.connectivity - 1.0.0.0 | Listening on port 5555
2015-08-21 11:25:36,752 | ERROR | rint Extender: 1 | CompositionManager               | 52 - core.caos - 1.0.0.0 | Unable to read new CompositionSpecification. Provided XML was ''.
javax.xml.bind.UnmarshalException
 - with linked exception:
[org.xml.sax.SAXParseException: Premature end of file.]
    at javax.xml.bind.helpers.AbstractUnmarshallerImpl.createUnmarshalException(AbstractUnmarshallerImpl.java:350)[:2.5.0]
    at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallerImpl.createUnmarshalException(UnmarshallerImpl.java:563)[:1.8.0_45]
    at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:249)[:1.8.0_45]
    at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:214)[:1.8.0_45]
    at javax.xml.bind.helpers.AbstractUnmarshallerImpl.unmarshal(AbstractUnmarshallerImpl.java:172)[:2.5.0]
    at javax.xml.bind.helpers.AbstractUnmarshallerImpl.unmarshal(AbstractUnmarshallerImpl.java:229)[:2.5.0]
    at eu.netide.core.caos.composition.CompositionSpecificationLoader.Load(CompositionSpecificationLoader.java:26)
    at eu.netide.core.caos.CompositionManager.setCompositionSpecificationXml(CompositionManager.java:117)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)[:1.8.0_45]
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)[:1.8.0_45]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)[:1.8.0_45]
    at java.lang.reflect.Method.invoke(Method.java:497)[:1.8.0_45]
    at org.apache.aries.blueprint.utils.ReflectionUtils$MethodPropertyDescriptor.internalSet(ReflectionUtils.java:628)[12:org.apache.aries.blueprint.core:1.4.3]
    at org.apache.aries.blueprint.utils.ReflectionUtils$PropertyDescriptor.set(ReflectionUtils.java:378)[12:org.apache.aries.blueprint.core:1.4.3]
    at org.apache.aries.blueprint.container.BeanRecipe.setProperty(BeanRecipe.java:939)[12:org.apache.aries.blueprint.core:1.4.3]
    at org.apache.aries.blueprint.container.BeanRecipe.setProperties(BeanRecipe.java:907)[12:org.apache.aries.blueprint.core:1.4.3]
    at org.apache.aries.blueprint.container.BeanRecipe.setProperties(BeanRecipe.java:888)[12:org.apache.aries.blueprint.core:1.4.3]
    at org.apache.aries.blueprint.container.BeanRecipe.internalCreate2(BeanRecipe.java:820)[12:org.apache.aries.blueprint.core:1.4.3]
    at org.apache.aries.blueprint.container.BeanRecipe.internalCreate(BeanRecipe.java:787)[12:org.apache.aries.blueprint.core:1.4.3]
    at org.apache.aries.blueprint.di.AbstractRecipe$1.call(AbstractRecipe.java:79)[12:org.apache.aries.blueprint.core:1.4.3]
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)[:1.8.0_45]
    at org.apache.aries.blueprint.di.AbstractRecipe.create(AbstractRecipe.java:88)[12:org.apache.aries.blueprint.core:1.4.3]
    at org.apache.aries.blueprint.container.BlueprintRepository.createInstances(BlueprintRepository.java:245)[12:org.apache.aries.blueprint.core:1.4.3]
    at org.apache.aries.blueprint.container.BlueprintRepository.createAll(BlueprintRepository.java:183)[12:org.apache.aries.blueprint.core:1.4.3]
    at org.apache.aries.blueprint.container.BlueprintContainerImpl.instantiateEagerComponents(BlueprintContainerImpl.java:682)[12:org.apache.aries.blueprint.core:1.4.3]
    at org.apache.aries.blueprint.container.BlueprintContainerImpl.doRun(BlueprintContainerImpl.java:377)[12:org.apache.aries.blueprint.core:1.4.3]
    at org.apache.aries.blueprint.container.BlueprintContainerImpl.run(BlueprintContainerImpl.java:269)[12:org.apache.aries.blueprint.core:1.4.3]
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)[:1.8.0_45]
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)[:1.8.0_45]
    at org.apache.aries.blueprint.container.ExecutorServiceWrapper.run(ExecutorServiceWrapper.java:106)[12:org.apache.aries.blueprint.core:1.4.3]
    at org.apache.aries.blueprint.utils.threading.impl.DiscardableRunnable.run(DiscardableRunnable.java:48)[12:org.apache.aries.blueprint.core:1.4.3]
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)[:1.8.0_45]
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)[:1.8.0_45]
    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)[:1.8.0_45]
    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)[:1.8.0_45]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_45]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_45]
    at java.lang.Thread.run(Thread.java:745)[:1.8.0_45]
Caused by: org.xml.sax.SAXParseException: Premature end of file.
    at org.apache.xerces.util.ErrorHandlerWrapper.createSAXParseException(Unknown Source)[:]
    at org.apache.xerces.util.ErrorHandlerWrapper.fatalError(Unknown Source)[:]
    at org.apache.xerces.impl.XMLErrorReporter.reportError(Unknown Source)[:]
    at org.apache.xerces.impl.XMLErrorReporter.reportError(Unknown Source)[:]
    at org.apache.xerces.impl.XMLErrorReporter.reportError(Unknown Source)[:]
    at org.apache.xerces.impl.XMLVersionDetector.determineDocVersion(Unknown Source)[:]
    at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)[:]
    at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)[:]
    at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)[:]
    at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source)[:]
    at org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)[:]
    at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:243)[:1.8.0_45]
    ... 35 more
2015-08-21 11:25:36,757 | WARN  | rint Extender: 1 | CompositionManager               | 52 - core.caos - 1.0.0.0 | Reusing previous specification ''.
</pre>

Is there something I'm missing?

Best,

Kévin

**Comment by KevinPhemius** [comment]
I'm using karaf 4.0.1

## 

Kévin

**Comment by TimVinkemeier** [comment]
Hi Kevin,

no, this is actually expected, since the core starts up with an empty compositionSpecification that of course cannot be parsed correctly. Therefore, it throws the error. I will include some code to prevent it from showing the stacktrace when the specification was empty.

It is expected that the compositionSpecification is set after startup using the Management Interface.

Best,
Tim

**Comment by andres-beato** [comment]
Hi!

I have the same error as you Kevin, when I try to follow the deployment steps:

karaf@root()> feature:repo-add mvn:eu.netide.core/core/1.0.0.0/xml/features
Adding feature url mvn:eu.netide.core/core/1.0.0.0/xml/features
karaf@root()> feature:install netide-core
DefaultValidationEventHandler: [FATAL_ERROR]: Premature end of file. 
     Location: 

Is it the solution published yet? And if it is so, which branch is it in?

Thanks in advance.

Best,
Andres Beato

**Comment by KevinPhemius** [comment]
Hi,

It's in the last version of CoreImplementation I think.

Better ask Tim.

Best,

Kévin

On Sep 1, 2015 7:12 PM, andres-beato notifications@github.com wrote:
Hi!

I have the same error as you Kevin, when I try to follow the deployment steps:

karaf@root()> feature:repo-add mvn:eu.netide.core/core/1.0.0.0/xml/features
Adding feature url mvn:eu.netide.core/core/1.0.0.0/xml/features
karaf@root()> feature:install netide-core
DefaultValidationEventHandler: [FATAL_ERROR]: Premature end of file.
     Location:

Is it the solution published yet? And if it is so, which branch is it in?

Thanks in advance.

Best,
Andres Beato

---

**Comment by TimVinkemeier** [comment]
As I already stated above, this is not an error.
It is just the output of the DefaultValidationEventHandler of JAXB trying to parse an invalid composition specification and encountering the end of the file unexpectedly.
However, to prevent any further confusion about this, I commented out the DVEH in 71faa2f, so this will not show anymore. Just pull the commit and use that version.

I hope this helps to resolve the Irritation :wink: 

Best,
Tim
